File: Sort/SelectionSort.js
/*
 * Problem: Selection sort
 * Author: Dung Nguyen Tien (Chris)
*/
function selectionSort(arr) {
  var imin = 0;
  for (var i = 0; i < arr.length - 1; i++) {
    imin = i;
    for (var j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[imin]) imin = j;
    }
    [arr[i], arr[imin]] = [arr[imin], arr[i]];
  }
}

var arr = [1,5,3,7,8,3, 0];
selectionSort(arr);
document.writeln(JSON.stringify(arr));

File: Sort/SelectionSort.cs
/*
 * Problem: SelectionSort
 * Author: Dung Nguyen Tien (Chris)
*/

public static class Sort
{
	public static void SelectionSort(List<int> arr)
	{
		var Length = arr.Count;
		var min = int.MaxValue;
		var imin = 0;
		
		for(var i = 0; i < Length - 1; i++)
		{
			min = arr[i];
			imin = i;

			for (var j = i + 1; j < Length; j++)
			{
				if (arr[j] < min)
				{
					min = arr[j];
					imin = j;
				}
			}

			swap(arr, i, imin);
		}
	}

	static void swap(List<int> arr, int i, int j)
	{
		if (i == j) return;

		var tmp = arr[i];
		arr[i] = arr[j];
		arr[j] = tmp;
	}
}